lib/tasks/one_time/2021-Q3/change_payments_expiry_to_2021-10-31.rake

```diff
@@ -0,0 +1,59 @@
+require 'active_support/logger'
+
+namespace :shf do
+
+  namespace :one_time do
+
+    def payments_to_change(time_range)
+      Payment.completed.where(expire_date: time_range).distinct.order(:expire_date)
+    end
+
+
+    def change_payment_expire_day_and_notes(payment, new_expire_date, task_name)
+      changed_by_task_str = "Changed by rake task #{task_name} on #{Time.now} : "
+      last_day_changed_str = "Payment expire_date changed to #{new_expire_date.iso8601}. Original expire_date was #{payment.expire_date.iso8601}."
+
+      new_note = payment.notes.to_s + ' | ' +
+         changed_by_task_str + last_day_changed_str
+      payment.update!(expire_date: new_expire_date)
+      payment.update!(notes: new_note)
+      payment
+    end
+
+
+    desc "change Payments expire_date to 2021-Oct-31 for members and companies with expire_date 2021-01-01 to 2021-10-30"
+    task change_payments_expire_date_to_2021_10_31: [:environment] do |this_task|
```

Comment:
I guess I was expecting that we would change the current `Membership` instance for a user, instead of adjusting the user's latest `Payment`.  (of course, we have to modify the `Payment` for h-branding).

Is this because we haven't fully extended the umbrella `Membership` everywhere it will eventually be (and thus, for instance, reminders to pay for next membership period are still driven off of payment expire date)?

Reply:
It's because none of the new `Membership` code has been deployed yet, but we need to deploy this ASAP.  So yes: `Payment expire date` is still the basis of membership period.